src/main/groovy/com/wooga/gradle/io/ProcessExecutor.groovy

```diff
@@ -0,0 +1,240 @@
+            withStandardError(spec.getStandardErrorStream(standardError, configure))
+        }
+        this
+    }
+
+    /**
+     * Sets the standard output and standard error streams to the same tile
+     * @param spec A trait for generating an output stream
+     * @param file The file to write standard output and standard error to
+     */
+    ProcessExecutor withOutput(OutputStreamSpec spec, File file, Action<OutputStreamConfiguration> configure = null) {
+        withOutput(spec, file, file, configure)
+    }
+
+    /**
+     * Sets the standard output and standard error streams
+     * @param outputStreamSpec A trait for generating an output stream
+     * @param stdout A trait for setting a log file
+     * @param standardError Whether to write standard error to the log file
+     */
+    ProcessExecutor withOutputLogFile(OutputStreamSpec outputStreamSpec,
+                                      LogFileSpec stdout,
+                                      LogFileSpec stderr,
+                                      Action<OutputStreamConfiguration> configure = null) {
+        withOutput(outputStreamSpec,
+            stdout.logFile.asFile.getOrNull(),
+            stderr.logFile.asFile.getOrNull(),
+            configure)
+    }
+
+    /**
+     * Writes the standard output and error to the same file
+     * @param outputStreamSpec A trait for generating an output stream
+     * @param logFile A trait containing a log file
+     */
+    ProcessExecutor withOutputLogFile(OutputStreamSpec outputStreamSpec,
+                                      LogFileSpec logFile,
+                                      Action<OutputStreamConfiguration> configure = null) {
+        withOutputLogFile(outputStreamSpec,
+            logFile,
+            logFile,
+            configure)
+    }
+
+    /**
+     * Sets the standard output stream
+     */
+    ProcessExecutor withStandardOutput(OutputStream stream) {
+        _standardOutput = stream
+        this
+    }
+
+    /**
+     * Sets the standard error stream
+     */
+    ProcessExecutor withStandardError(OutputStream stream) {
+        _standardError = stream
+        this
+    }
+
+    /**
+     * Executes the specified executable according to the configuration
+     * @return The result of the execution
+     */
+    ExecResult execute() {
+        function.apply({ exec ->
+            exec.with {
+                executable _executable
+
+                if (workingDirectory) {
+                    workingDir(workingDirectory)
+                }
+
+                if (arguments != null) {
+                    it.args = arguments
+                }
+
+                if (environment != null) {
+                    if (appendToEnvironment) {
+                        it.environment(environment)
+                    } else {
+                        it.environment = environment
+                    }
+                }
+
+                if (_standardOutput != null) {
```

Comment:
If the underline is because of these conflicts, i believe you can solve them using an explicity `this` reference to the class attributes.

Reply:
If it doesn't work (groovy variable scoping can be very annoying sometimes) you can create these underlined variables in a local context, before the `function.apply` call.